### PR TITLE
Mosquito spawn random

### PR DESCRIPTION
**Pull request by nlaneau**
Includes code from PR Many-Mosquito.

Created SpawnMosquito function that sets a new mosquito's coordinates to a random spot, assuming X is 0-800 and Y is 0-600.

### CODE REVIEW

File: MJ68-sleep.ict [hunk 1]
```diff
@@ -462,12 +463,12 @@ types:
       // For debugging
       //ct.types.copy('mosquito', ct.random.range(800), ct.random.range(600));
 
-      this.getRoom().won = true;
+      this.kill = true;
```
Comment (annedo) [inline MJ68-sleep.ict]: Delete all this, `kill` is a default parameter of every new copy

File: MJ68-sleep.ict [hunk 2]
```diff
@@ -478,13 +478,60 @@ rooms:
...
       this.won = false;
+      this.mosquitoCounter = 0;
+      this.maxMosquito = 5;
+
+      // Returns a mosquito that is set to spawn along the perimeter of the room
+      this.SpawnMosquito = (() => {
+          var edgeSpawn = Math.floor(ct.random.range(0, 4)); // Random screen edge for the spawn
+
+          var maxWidth = 800; // TODO - get this value from properties
+          var maxHeight = 600; // TODO - get this value from properties
```
Comment (annedo) [inline MJ68-sleep.ict]: Use `ct.camera.width` and `ct.camera.height`

File: MJ68-sleep.ict [hunk 3]
```diff
@@ -499,52 +546,66 @@ rooms:
...
-          ct.backgrounds.add('furrowed', 0, -9);
-      }
+          if (this.mosquito.kill && !this.won) {
+              this.mosquitoCounter++;               
+          }
 
-      if (this.gameTimer.timeLeft < this.initTime/6) {
-          ct.backgrounds.add('lose', 0, -8);
-      }
+          console.log(this.mosquitoCounter + " " + this.maxMosquito);
```
Comment (annedo) [inline MJ68-sleep.ict]: Remove for final build

File: MJ68-sleep.ict [hunk 4]
```diff
@@ -499,52 +546,66 @@ rooms:
...
-          this.arm.scale.y = 1;
-      }
+          if (this.gameTimer.timeLeft == 0 || this.mosquito.kill) {
+              if (this.won) {
+                  ct.backgrounds.add('happy', 0, -7); 
+              }
+              else {
+                  ct.backgrounds.add('lose', 0, -8);
+              }        
+          }
```
Comment (annedo) [inline MJ68-sleep.ict]: You're already fixing this, thanks

**Review by annedo — CHANGES_REQUESTED** [review]

File: MJ68-sleep.ict (hunk 1 above)
Comment (nlaneau) [reply MJ68-sleep.ict]: You're right, this `this.kill` gets updated in the OnStep of the main room anyway.

Comment (nlaneau) [reply MJ68-sleep.ict]: Fixed in [d038378](https://github.com/annedo/Minijam68-sleep/pull/2/commits/d038378399a1b607341f7a75908fdaab5217cb83)

File: MJ68-sleep.ict (hunk 2 above)
Comment (nlaneau) [reply MJ68-sleep.ict]: Fixed in [d038378](https://github.com/annedo/Minijam68-sleep/pull/2/commits/d038378399a1b607341f7a75908fdaab5217cb83)

File: MJ68-sleep.ict (hunk 3 above)
Comment (nlaneau) [reply MJ68-sleep.ict]: Fixed in [d038378](https://github.com/annedo/Minijam68-sleep/pull/2/commits/d038378399a1b607341f7a75908fdaab5217cb83)

File: MJ68-sleep.ict (hunk 4 above)
Comment (nlaneau) [reply MJ68-sleep.ict]: Fixed in [d038378](https://github.com/annedo/Minijam68-sleep/pull/2/commits/d038378399a1b607341f7a75908fdaab5217cb83)

**Pull request merged**